File: queue.cs
using System;

class LinkedListNode
{
    public int Data;
    public LinkedListNode Next;

    public LinkedListNode(int data)
    {
        Data = data;
        Next = null;
    }
}

class Queue
{
    private LinkedListNode Front;
    private LinkedListNode Back;

    public void Enqueue(int data)
    {
        LinkedListNode node = new LinkedListNode(data);

        if (Back == null)
        {
            Front = node;
            Back = node;
        }
        else
        {
            Back.Next = node;
            Back = node;
        }
    }

    public int Dequeue()
    {
        if (Front == null)
        {
            throw new InvalidOperationException("Queue is empty.");
        }

        int data = Front.Data;
        Front = Front.Next;

        if (Front == null)
        {
            Back = null;
        }

        return data;
    }
}

class Program
{
    static void Main()
    {
        Queue queue = new Queue();

        queue.Enqueue(20);
        queue.Enqueue(30);

        Console.WriteLine("Dequeued: {0}", queue.Dequeue());
        Console.WriteLine("Dequeued: {0}", queue.Dequeue());
    }
}

File: queue.c
#include <stdio.h>
#include <stdlib.h>

struct LinkedListNode {
    int data;
    struct LinkedListNode* next;
};

struct Queue {
    struct LinkedListNode* front;
    struct LinkedListNode* back;
};

struct LinkedListNode* createNode(int data) {
    struct LinkedListNode* newNode = (struct LinkedListNode*)malloc(sizeof(struct LinkedListNode));
    newNode->data = data;
    newNode->next = NULL;
    return newNode;
}

void enqueue(struct Queue* queue, int data) {
    struct LinkedListNode* node = createNode(data);

    if (queue->back == NULL) {
        queue->front = node;
        queue->back = node;
    } else {
        queue->back->next = node;
        queue->back = node;
    }
}

int dequeue(struct Queue* queue) {
    if (queue->front == NULL) {
        fprintf(stderr, "Queue is empty.\n");
        exit(EXIT_FAILURE);
    }

    int data = queue->front->data;
    struct LinkedListNode* temp = queue->front;
    queue->front = queue->front->next;

    free(temp);

    if (queue->front == NULL) {
        queue->back = NULL;
    }

    return data;
}

int main() {
    struct Queue queue;
    queue.front = NULL;
    queue.back = NULL;

    enqueue(&queue, 20);
    enqueue(&queue, 30);

    printf("Dequeued: %d\n", dequeue(&queue));
    printf("Dequeued: %d\n", dequeue(&queue));

    return 0;
}
